tools/deviceGen/output/traits_defaults.hlsl

```diff
@@ -0,0 +1,382 @@
+NBL_GENERATE_GET_OR_DEFAULT(maxFragmentCombinedOutputResources, uint32_t, 16);
+NBL_GENERATE_GET_OR_DEFAULT(maxComputeSharedMemorySize, uint32_t, 32768);
+NBL_GENERATE_GET_OR_DEFAULT(maxComputeWorkGroupCount[3], uint32_t, {MinMaxWorkgroupCount,MinMaxWorkgroupCount,MinMaxWorkgroupCount});
+NBL_GENERATE_GET_OR_DEFAULT(maxComputeWorkGroupInvocations, uint16_t, MinMaxWorkgroupInvocations);
+NBL_GENERATE_GET_OR_DEFAULT(maxWorkgroupSize[3], uint16_t, {MinMaxWorkgroupInvocations,MinMaxWorkgroupInvocations,64u});
+NBL_GENERATE_GET_OR_DEFAULT(subPixelPrecisionBits, uint8_t, 4);
+NBL_GENERATE_GET_OR_DEFAULT(subTexelPrecisionBits, uint8_t, 4);
+NBL_GENERATE_GET_OR_DEFAULT(mipmapPrecisionBits, uint8_t, 4);
+NBL_GENERATE_GET_OR_DEFAULT(maxDrawIndirectCount, uint32_t, 1073741824);
+NBL_GENERATE_GET_OR_DEFAULT(maxSamplerLodBias, float, 4);
+NBL_GENERATE_GET_OR_DEFAULT(maxSamplerAnisotropyLog2, uint8_t, 4);
+NBL_GENERATE_GET_OR_DEFAULT(maxViewports, uint8_t, 16);
+NBL_GENERATE_GET_OR_DEFAULT(maxViewportDims[2], uint16_t, {MinMaxImageDimension2D,MinMaxImageDimension2D});
+NBL_GENERATE_GET_OR_DEFAULT(viewportBoundsRange[2], float, { -MinMaxImageDimension2D*2u, MinMaxImageDimension2D*2u-1 });
+NBL_GENERATE_GET_OR_DEFAULT(viewportSubPixelBits, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(minMemoryMapAlignment, uint16_t, 64);
+NBL_GENERATE_GET_OR_DEFAULT(bufferViewAlignment, uint16_t, 64);
+NBL_GENERATE_GET_OR_DEFAULT(minUBOAlignment, uint16_t, 256);
+NBL_GENERATE_GET_OR_DEFAULT(minSSBOAlignment, uint16_t, 64);
+NBL_GENERATE_GET_OR_DEFAULT(minTexelOffset, int8_t, -8);
+NBL_GENERATE_GET_OR_DEFAULT(maxTexelOffset, uint8_t, 7);
+NBL_GENERATE_GET_OR_DEFAULT(minTexelGatherOffset, int8_t, -8);
+NBL_GENERATE_GET_OR_DEFAULT(maxTexelGatherOffset, uint8_t, 7);
+NBL_GENERATE_GET_OR_DEFAULT(minInterpolationOffset, float, -0.5);
+NBL_GENERATE_GET_OR_DEFAULT(maxInterpolationOffset, float, 0.437500000);
+NBL_GENERATE_GET_OR_DEFAULT(subPixelInterpolationOffsetBits, uint8_t, MinSubPixelInterpolationOffsetBits);
+NBL_GENERATE_GET_OR_DEFAULT(maxFramebufferWidth, uint32_t, MinMaxImageDimension2D);
+NBL_GENERATE_GET_OR_DEFAULT(maxFramebufferHeight, uint32_t, MinMaxImageDimension2D);
+NBL_GENERATE_GET_OR_DEFAULT(maxFramebufferLayers, uint32_t, 1024);
+NBL_GENERATE_GET_OR_DEFAULT(maxColorAttachments, uint8_t, MinMaxColorAttachments);
+NBL_GENERATE_GET_OR_DEFAULT(maxSampleMaskWords, uint8_t, 1);
+NBL_GENERATE_GET_OR_DEFAULT(timestampPeriodInNanoSeconds, float, 83.334);
+NBL_GENERATE_GET_OR_DEFAULT(maxClipDistances, uint8_t, 8);
+NBL_GENERATE_GET_OR_DEFAULT(maxCullDistances, uint8_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxCombinedClipAndCullDistances, uint8_t, 8);
+NBL_GENERATE_GET_OR_DEFAULT(discreteQueuePriorities, uint32_t, 2);
+NBL_GENERATE_GET_OR_DEFAULT(pointSizeRange[2], float, {1.f,64.f});
+NBL_GENERATE_GET_OR_DEFAULT(lineWidthRange[2], float, {1.f,1.f});
+NBL_GENERATE_GET_OR_DEFAULT(pointSizeGranularity, float, 1);
+NBL_GENERATE_GET_OR_DEFAULT(lineWidthGranularity, float, 1);
```

Comment:
you might want to pay attention to print `.f` suffices on floats, so that the compiler isn't screaming warnings at us that we're casting int to float.

Generally you should test that the trait classes and JIT headers both compile with DXC and that DXC emits no warnings.

Reply:
still missing sufficies on most `float`

Reply:
hmm I guess the `float` type needs to change to `uint32_t` and the correctly suffixed float literals need `asuint()` around them